src/main/resources/db/migration/0300/V338__BiomassSpecies.sql

```diff
@@ -0,0 +1,62 @@
+CREATE TABLE tracking.observation_biomass_species (
+    id BIGINT PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    observation_id BIGINT NOT NULL REFERENCES tracking.observations ON DELETE CASCADE,
+    monitoring_plot_id BIGINT NOT NULL REFERENCES tracking.monitoring_plots ON DELETE CASCADE,
+    species_id BIGINT REFERENCES species,
+    scientific_name TEXT,
+    common_name TEXT,
+    is_invasive BOOLEAN NOT NULL,
+    is_threatened BOOLEAN NOT NULL,
+
+    CONSTRAINT species_identifier
+        CHECK ((species_id IS NOT NULL AND scientific_name IS NULL)
+            OR (species_id IS NULL AND scientific_name IS NOT NULL)),
+
+    FOREIGN KEY (observation_id, monitoring_plot_id)
+        REFERENCES tracking.observation_biomass_details (observation_id, monitoring_plot_id)
+        ON DELETE CASCADE,
+
+    UNIQUE(id, observation_id, monitoring_plot_id)
+);
+
+CREATE INDEX ON tracking.observation_biomass_species(observation_id);
+CREATE INDEX ON tracking.observation_biomass_species(monitoring_plot_id);
+
+CREATE UNIQUE INDEX ON tracking.observation_biomass_species
+    (observation_id, monitoring_plot_id, species_id)
+    WHERE species_id IS NOT NULL;
+
+CREATE UNIQUE INDEX ON tracking.observation_biomass_species
+    (observation_id, monitoring_plot_id, scientific_name)
+    WHERE scientific_name IS NOT NULL;
+
+-- This table is now redundant
+DROP TABLE tracking.observation_biomass_additional_species;
+
+DROP TABLE tracking.observation_biomass_quadrat_species;
+CREATE TABLE tracking.observation_biomass_quadrat_species (
+    observation_id BIGINT NOT NULL REFERENCES tracking.observations ON DELETE CASCADE,
+    monitoring_plot_id BIGINT NOT NULL REFERENCES tracking.monitoring_plots ON DELETE CASCADE,
+    position_id INTEGER NOT NULL REFERENCES tracking.observation_plot_positions,
+    biomass_species_id BIGINT NOT NULL REFERENCES tracking.observation_biomass_species ON DELETE CASCADE,
+    abundance_percent INTEGER NOT NULL CHECK ( abundance_percent BETWEEN 0 AND 100),
+
+    FOREIGN KEY (observation_id, monitoring_plot_id, biomass_species_id)
+        REFERENCES tracking.observation_biomass_species (observation_id, monitoring_plot_id, id)
+        ON DELETE CASCADE,
+
+    UNIQUE (observation_id, monitoring_plot_id, position_id, biomass_species_id)
```

Comment:
Could make this `PRIMARY KEY` instead of `UNIQUE` to slightly clarify the intent.